corehq/apps/reports/views.py

```diff
@@ -1756,7 +1756,11 @@ def _error(msg):
         instance = _get_location_safe_form(domain, request.couch_user, instance_id)
         context = _get_form_context(request, domain, instance)
         if not instance.app_id or not instance.build_id:
-            return _error(_('Could not detect the application/form for this submission.'))
+            return _error(_(
+                "This submission was made from the App Preview."
+                "As a result it is not possible to edit this submission\n\n"
+                "If this was not submitted from App Preview, please report an issue"
```

Comment:
Yea I think App Preview is the main use case, but certainly not the only. Form Preview also creates un-editable forms, as does any sort of form submitted to commcare via api that isn't attached to an app. guessing system forms act similarly.

We could say "Could not detect the application/form for this submission. A common cause is that the form was submitted via App or Form preview." or something like that. i like the idea of checking the deviceID